packages/runtime/src/runtime.ts

```diff
@@ -471,80 +500,60 @@ export class Runtime {
             fromAccount.closeApp(txnParam.appId); // remove app from local state
             break;
           }
-          case TransactionType.ClearSSC: {
-            // https://developer.algorand.org/docs/reference/cli/goal/app/clear/#search-overlay
-            fromAccount.closeApp(txnParam.appId); // remove app from local state
-            break;
-          }
         }
       }
     });
   }
 
-  // get execution mode (stateless or application) based on txParams
-  getExecutionMode (txnParam: ExecParams): ExecutionMode {
-    if (txnParam.sign === SignType.LogicSignature) {
-      return ExecutionMode.STATELESS;
-    }
-    return ExecutionMode.STATEFUL;
-  }
-
   /**
    * This function executes a transaction based on a smart
    * contract logic and updates state afterwards
    * @param txnParams : Transaction parameters
    * @param program : teal code as a string
    * @param args : external arguments to smart contract
    */
-  /* eslint-disable sonarjs/cognitive-complexity */
-  executeTx (txnParams: ExecParams | ExecParams[], program: string,
-    args: Uint8Array[]): void {
+  executeTx (txnParams: ExecParams | ExecParams[], args?: Uint8Array[]): void {
     const [tx, gtxs] = this.createTxnContext(txnParams); // get current txn and txn group (as encoded obj)
 
     // initialize context before each execution
     this.ctx = {
       state: cloneDeep(this.store), // state is a deep copy of store
       tx: tx,
       gtxs: gtxs,
-      args: args
+      args: args ?? []
```

Comment:
Why do we need args now? I don't think external arguments are passed to stateful smart contracts. And args for stateless should be inside logic signature. Right?

Reply:
i thought it has some logic, are you sure about this ? i will remove

Reply:
just checked we are using these args in args opcode

Reply:
that means we have to keep this in logicSig part.

Reply:
Exactly. For stateless external arguments (using `Arg` op) are passed in logicSig, and for stateful we have application args (in `txParams.payflags.appArgs`)